src/components/FlowRunGraph.vue

```diff
@@ -0,0 +1,382 @@
+<template>
+  <div v-if="(layers.length > 0)" class="flow-run-graph">
+    <FlowRunGraphPixiRenderer :layers="layers" :graph-data="graphData" />
+  </div>
+</template>
+
+<script lang="ts" setup>
+  import { useSubscription } from '@prefecthq/vue-compositions'
+  import { computed, watchEffect } from 'vue'
+  import FlowRunGraphPixiRenderer from '@/components/FlowRunGraphPixiRenderer.vue'
+  import { useWorkspaceApi } from '@/compositions'
+  import {
+    ModifiedGraphNode,
+    ModifiedGraphData,
+    Layers
+  } from '@/models'
+
+  const props = defineProps<{
+    flowRunId: string,
+  }>()
+
+  const api = useWorkspaceApi()
+
+  const graphSubscription = useSubscription(api.flowRuns.getFlowRunsGraph, [props.flowRunId]) // { interval:  5000 }
+  const graphData = computed(() => {
+    const data: ModifiedGraphData = [...graphSubscription.response ?? []]
+
+    // fill in downstreamDependencies from upstreamDependencies
+    // @TODO this should be done on the backend
+    // after, sort by startTime
+    const formattedData = data.map(node => {
+      const nodeClone = { ...node }
+      if (
+        nodeClone.upstreamDependencies.length === 1
+        && nodeClone.upstreamDependencies[0].id === props.flowRunId
+      ) {
+        nodeClone.upstreamDependencies = []
+      }
+
+      nodeClone.upstreamDependencies.forEach(parentDependency => {
+        const parentNode = data.find(dataNode => dataNode.id === parentDependency.id)
+        if (!parentNode) {
+          console.error('Upstream dependency node not found', parentDependency)
+          return
+        }
+
+        if (!parentNode.downstreamDependencies) {
+          parentNode.downstreamDependencies = []
+        }
+        parentNode.downstreamDependencies.push(nodeClone.id)
+      })
+
+      return nodeClone
+    }).sort((a, b) => {
+      if (a.startTime && b.startTime) {
+        return a.startTime.getTime() > b.startTime.getTime() ? 1 : -1
+      }
+      return 0
+    })
+
+    return formattedData
+  })
+
+  watchEffect(() => {
+    console.log('DATA', graphData.value)
+  })
+
+  type LayersEngine = {
+    layers: Layers,
+    resetLayers: () => void,
+    generateLayersFromRootNode: (
+      rootNode: ModifiedGraphNode,
+      nodeData: ModifiedGraphData
+    ) => void,
+    sortLayerNodesIntoRows: (nodeData: ModifiedGraphData) => void,
+    argueOverlaps: (
+      layerIndex: number,
+      nodeData: ModifiedGraphData,
+      nodeIds: string[],
+    ) => void,
+    shoveUntilLayer: (
+      sourceLayerIndex: number,
+      sourceRow: number,
+    ) => void,
+    shoveLayer: (
+      sourceLayerIndex: number,
+      sourceRow: number,
+    ) => void,
+  }
+  const layersEngine: LayersEngine = {
```

Comment:
Note that layer related stuff is not used here. It was created for use in the DAG view.